metricflow/test/naming/test_dunder_naming_scheme.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+
+from typing import Sequence
+
+import pytest
+from dbt_semantic_interfaces.references import EntityReference
+from dbt_semantic_interfaces.type_enums import TimeGranularity
+from dbt_semantic_interfaces.type_enums.date_part import DatePart
+
+from metricflow.naming.dunder_scheme import DunderNamingScheme
+from metricflow.specs.specs import DimensionSpec, EntitySpec, LinkableInstanceSpec, TimeDimensionSpec
+from metricflow.test.time.metric_time_dimension import MTD_SPEC_MONTH, MTD_SPEC_WEEK, MTD_SPEC_YEAR
+
+
+@pytest.fixture(scope="session")
+def dunder_naming_scheme() -> DunderNamingScheme:  # noqa: D
+    return DunderNamingScheme()
+
+
+def test_input_str(dunder_naming_scheme: DunderNamingScheme) -> None:  # noqa: D
```

Comment:
nit: I'm not a fan of tests that do multi-step assertions against different logic triggers, I'd rather they be parameterized on input/output or else split into separate methods - that way when I inevitably change something that breaks a bunch of stuff I don't have to go through this `run test, fix bug one, run test, fix bug 2, run test, fix bug 3` workflow because all of the errors are exposed on the first run.